src/App.tsx

```diff
@@ -1,61 +1,156 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+const getUser = (userId: number): User | null => (
+  usersFromServer.find(user => userId === user.id) || null
+);
+
+const getValidId = (todos: Todo[]): number => {
+  const existingIds = todos.map(todo => todo.id);
+
+  return Math.max(...existingIds) + 1;
+};
+
+const todosWithUser: Todo[] = todosFromServer
+  .map(todo => ({
+    ...todo,
+    user: getUser(todo.userId),
+  }));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [hasNoTitleError, setHasNoTitleError] = useState(false);
+  const [userNotSelectedError, setUserNotSelectedError] = useState(false);
+
+  const clearForm = () => {
+    setTitle('');
+    setSelectedUser(0);
+  };
+
+  const handleUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserNotSelectedError(false);
+
+    setSelectedUser(+event.currentTarget.value);
```

Comment:
```suggestion
    setSelectedUser(+event.target.value);
```

https://www.jstips.co/en/javascript/difference-between-target-and-currentTarget/